tests/pytests/test.py

```diff
@@ -3497,3 +3497,39 @@ def test_empty_field_name(env):
     env.expect('FT.CREATE', 'idx', 'SCHEMA', '', 'TEXT').ok()
     conn.execute_command('hset', 'doc1', '', 'foo')
     env.expect('FT.SEARCH', 'idx', 'foo').equal([1L, 'doc1', ['', 'foo']])
+
+
+def test_clear_fields(env):
```

Comment:
How long does it take? Maybe run it on nightly? Or maybe add it to the performance tests?

Reply:
Added module param and made a test that runs twice with and without freeing resources on the main thread.
The test checks the freeing on a 2nd thread is more than 10 times faster.